FNF_MissionTemplate.VR/server/damage/fn_handleDamage.sqf

```diff
@@ -0,0 +1,44 @@
+/*
+ * Author: tanaKa
+ * Function used to handle random damage on vehicle destruction
+ * for FNF
+ *
+ * Params: 
+ * 0: vehicle <OBJECT>
+ *
+ * Return value:
+ * N/A
+ *
+ * Usage: 
+ * [vehicle] spawn PHX_fnc_handleDamage
+ *
+ * Public: no
+ */
+params ["_vehicle"];
+
+{
+	_kill = random [1,50,100];
+
+	if (_kill > 69) then {
+		_bodyPart = selectRandom ["Head", "Body", "LeftArm", "RightArm", "LeftLeg", "RightLeg"];
+		_damage = 1; // full damage
+		_type = selectRandom ["bullet", "vehiclecrash", "collision", "grenade", "explosive", "shell", "backblast", "stab", "unknown"];
+		[_x, _damage, _bodyPart, _type] call ace_medical_fnc_addDamageToUnit;
+		_x setDamage 1; // kill player
+		diag_log format ["Vehicle DEBUG: %1 damage added to %2's %3 with a %4 value with _kill set to %5",_type,(name player),_bodypart,_damage,_kill];
+	} else {
+		_damage = selectRandom [0,0,0,0,0,0.2,0.3,0.4]; //small chance to be slightly injured in a limb
+		if (_damage > 0) then {
+			_bodyPart = selectRandom ["LeftArm", "RightArm", "LeftLeg", "RightLeg"];
+			_type = selectRandom ["bullet", "vehiclecrash", "collision", "grenade", "explosive", "shell", "backblast", "stab", "unknown"];
+			[_x, _damage, _bodyPart, _type] call ace_medical_fnc_addDamageToUnit;
+			diag_log format ["Vehicle DEBUG: %1 damage added to %2's %3 with a %4 value with _kill set to %5",_type,(name player),_bodypart,_damage,_kill];
+			uiSleep 1;
+			[speed _vehicle isEqualTo 0, (moveOut _x)] call CBA_fnc_waitUntilAndExecute;
```

Comment:
Syntax here for `CBA_fnc_waitUntilAndExecute` would look like:
```
[{speed _this isEqualTo 0}, {
  {moveOut _x} forEach (crew _this);
}, _vehicle] call CBA_fnc_waitUntilAndExecute;
```
`_vehicle` doesn't exist within the scope of the CBA function, so you have to pass it as an argument.

I recommend changing the condition to something like this to cover more situations and adding a timeout:
```
[{abs (speed _this) < 0.1}, {
  {moveOut _x} forEach (crew _this);
}, _vehicle, 20] call CBA_fnc_waitUntilAndExecute;
```

Can change 0.1 and the timeout of 20 to whatever. This will ensure that the speed doesn't have to be exactly 0, which may not always be the case, and has a timeout of 20 seconds so this doesn't run forever if the condition won't be met for some reason.

Reply:
And lines 36,37,39,40 can be combined into 2 lines outside of and just after the scope of the 'if' statement here, as it will be executed regardless

Reply:
https://github.com/FridayNightFight/FNF/pull/100/commits/db0fd7c843d77ee4fefc75f3ee9f6b839076fe78 Changed